### PR TITLE
WT-12568 Fix s_all under GNU grep >= 3.8

### PR DESCRIPTION
**Pull request by korteland**
PR patch build has been extended to run `s_all` on every platform. I tested this on my Evergreen workstation with GNU grep 3.7, macOS with BSD grep 2.6, and my personal machine with GNU grep 3.11.

I'm not anticipating any problems here - in fact, `s_void` was already using `-E`, so I don't think it was incompatible anywhere we care about.

I also considered combining the short options (e.g. `grep -E -x` -> `grep -Ex`) but it didn't do anything for readability and these scripts aren't code golf.

### CODE REVIEW

File: dist/s_python [hunk 1]
```diff
@@ -7,7 +7,7 @@ cd_top
 
 # Check Python coding standards: check for tab characters.
 # Ignore generated files.
-egrep '	' `git ls-files -- '*.py'` |
+grep -E '	' `git ls-files -- '*.py'` |
```
[the remark this answers is not in the document]
Comment (korteland) [reply dist/s_python:10]: Thanks, but I actually didn't change this to a tab :-) It just displays differently now that there are two more characters at the start of the line.

You're right about the `skip $1` change - that was looking for a hard tab, but `copyright.list` uses spaces, so I altered it to just match any sort of gap.

[the remark this answers is not in the document]
Comment (ershov) [reply dist/s_python:10]: Can you replace the `'<tab>'` with `$'\t'` for clarity?

**Review by ershov — APPROVED** [review]
Thanks Will for the PR.

I like your decision **not** to merge the options but have them as separate arguments.

File: dist/s_python (hunk 1 above)
[the remark this answers is not in the document]
Comment (ershov) [reply dist/s_python:10]: ... or actually let's make it a separate ticket.

[the remark this answers is not in the document]
Comment (ershov) [reply dist/s_python:10]: WT-12866

[the remark this answers is not in the document]
Comment (korteland) [reply dist/s_python:10]: Thanks @ershov!

**Pull request merged**